File: src/test/java/it/marketplace/microservices/common/resource/PaymentInstallmentsResourceTest.java
package it.marketplace.microservices.common.resource;

import it.marketplace.microservices.common.enums.StatusOrderEnum;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;

import static org.junit.jupiter.api.Assertions.assertEquals;

class PaymentInstallmentsResourceTest {
    @Test
    void shouldCreatePaymentInstallmentsResourceWithAllFields() {
        // Arrange
        LocalDateTime tmsUpdate = LocalDateTime.now();
        // Act
        PaymentInstallmentsResource resource = new PaymentInstallmentsResource(1L, "REF123", "ORD123", StatusOrderEnum.PAID, 100.0, tmsUpdate);
        // Assert
        assertEquals(1L, resource.getId());
        assertEquals("REF123", resource.getReference());
        assertEquals("ORD123", resource.getOrderCode());
        assertEquals(StatusOrderEnum.PAID, resource.getStatus());
        assertEquals(100.0, resource.getDebit());
        assertEquals(tmsUpdate, resource.getTmsUpdate());
    }

    @Test
    void shouldSetAndGetIdAndTmsUpdate() {
        // Arrange
        PaymentInstallmentsResource resource = new PaymentInstallmentsResource();
        LocalDateTime tmsUpdate = LocalDateTime.now();
        // Act
        resource.setId(42L);
        resource.setTmsUpdate(tmsUpdate);
        // Assert
        assertEquals(42L, resource.getId());
        assertEquals(tmsUpdate, resource.getTmsUpdate());
    }
}

File: src/test/java/it/marketplace/microservices/common/resource/PaymentInstallmentsResourceTest.kt
package it.marketplace.microservices.common.resource

import it.marketplace.microservices.common.enums.StatusOrderEnum
import org.junit.jupiter.api.Test
import java.time.LocalDateTime
import kotlin.test.assertEquals

class PaymentInstallmentsResourceTest {
    @Test
    fun shouldCreatePaymentInstallmentsResourceWithAllFields() {
        // Arrange
        val tmsUpdate = LocalDateTime.now()
        // Act
        val resource = PaymentInstallmentsResource(1L, "REF123", "ORD123", StatusOrderEnum.PAID, 100.0, tmsUpdate)
        // Assert
        assertEquals(1L, resource.id)
        assertEquals("REF123", resource.reference)
        assertEquals("ORD123", resource.orderCode)
        assertEquals(StatusOrderEnum.PAID, resource.status)
        assertEquals(100.0, resource.debit)
        assertEquals(tmsUpdate, resource.tmsUpdate)
    }

    @Test
    fun shouldSetAndGetIdAndTmsUpdate() {
        // Arrange
        val resource = PaymentInstallmentsResource(
            id = null,
            reference = "REF123",
            orderCode = "ORD123",
            status = StatusOrderEnum.PAID,
            debit = 100.0,
            tmsUpdate = LocalDateTime.now()
        )
        val tmsUpdate = LocalDateTime.now()
        // Act
        resource.id = 42L
        resource.tmsUpdate = tmsUpdate
        // Assert
        assertEquals(42L, resource.id)
        assertEquals(tmsUpdate, resource.tmsUpdate)
    }
}

